### PR TITLE
PropagationConfiguration docstring + validation

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes a part of #2004 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > UTs pass
* [x] If applicable, add screenshots or log transcripts of the feature working
image

### CODE REVIEW

File: monkey/common/agent_configuration/agent_sub_configuration_schemas.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def _make_exploitation_options_configuration(self, data, **kwargs):
 
 
 class PropagationConfigurationSchema(Schema):
-    maximum_depth = fields.Int()
+    maximum_depth = fields.Int(validate=validate.Range(min=1))
```
Comment (mssalvatore) [inline monkey/common/agent_configuration/agent_sub_configuration_schemas.py:138]: Depth could be 0. The agent would run everything except the propagation phase.

File: monkey/common/agent_configuration/agent_sub_configurations.py [hunk 2]
```diff
@@ -118,6 +118,15 @@ class ExploitationConfiguration:
 
 @dataclass(frozen=True)
 class PropagationConfiguration:
+    """
+    A configuration for propagation
+
+    Attributes:
+        :param maximum_depth: Maximum number of hops to take from the Island server
```
Comment (mssalvatore) [inline monkey/common/agent_configuration/agent_sub_configurations.py:125]: I think we can do a better job describing this. Take a look in the documentation and see what we have there.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/common/agent_configuration/agent_sub_configuration_schemas.py (hunk 1 above)
Comment (shreyamalviya) [reply monkey/common/agent_configuration/agent_sub_configuration_schemas.py:138]: Thought the same but the UI's schema says 1 is the minimum. I'll change both.

File: monkey/common/agent_configuration/agent_sub_configurations.py (hunk 2 above)
Comment (shreyamalviya) [reply monkey/common/agent_configuration/agent_sub_configurations.py]: How about now?

File: monkey/monkey_island/cc/ui/src/services/configuration/propagation/propagation.js [hunk 3]
```diff
@@ -11,14 +11,15 @@ const PROPAGATION_CONFIGURATION_SCHEMA = {
...
+      'minimum': 0,
       'default': 2,
       'description': 'Amount of hops allowed for the monkey to spread from the ' +
       'Island server. \n' +
       ' \u26A0' +
       ' Note that setting this value too high may result in the ' +
       'Monkey propagating too far, '+
-      'if the "Local network scan" is enabled'
+      'if "Local network scan" is enabled.\n' +
+      'If the value is set to 0, everything except the propagation phase will run.'
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/services/configuration/propagation/propagation.js:22]:
```suggestion
      'if "Local network scan" is enabled.\n'
```
I would drop this. "Propagation phase" isn't well defined in the UI. You could Say "Setting this to 0 will disable all scanning and exploitation.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**